controllers/autoneg.go

```diff
@@ -20,16 +20,22 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"time"
 
+	"github.com/cenkalti/backoff"
 	"google.golang.org/api/compute/v1"
 	"google.golang.org/api/googleapi"
 )
 
 const (
-	autonegAnnotation       = "anthos.cft.dev/autoneg"
-	autonegStatusAnnotation = "anthos.cft.dev/autoneg-status"
-	negStatusAnnotation     = "cloud.google.com/neg-status"
-	autonegFinalizer        = "anthos.cft.dev/autoneg"
+	autonegAnnotation             = "anthos.cft.dev/autoneg"
+	autonegStatusAnnotation       = "anthos.cft.dev/autoneg-status"
+	negStatusAnnotation           = "cloud.google.com/neg-status"
+	autonegFinalizer              = "anthos.cft.dev/autoneg"
+	computeOperationStatusDone    = "DONE"
+	computeOperationStatusRunning = "RUNNING"
+	computeOperationStatusPending = "PENDING"
+	maxElapsedTime                = 15 * time.Minute
```

Comment:
Let's use a lower timeout.

The impact of service-reconcile requests piling up in the (admittedly unlikely) case of a very long timeout I think is much higher than just resubmitting a long-running operation.

For Traffic Director, we can expect that these operations *should* complete in about a minute.
The terraform timeout is four minutes giving us a real-world upper bound.

Shall we use four minutes, or did you have another reason to choose a higher timeout?

Reply:
4 Minutes is fine to me, we can always change that if we hit the upper limit.

Looking into where does 4 minutes (because they seemed oddly specific to me) came from:
 https://github.com/hashicorp/terraform-provider-google/pull/3345 and https://github.com/hashicorp/terraform-provider-google/blame/919e9aef4d253081219a3f032246e0506904323e/google/resource_compute_backend_service.go#L183-L185
or https://github.com/GoogleCloudPlatform/magic-modules/pull/1978

But not really clear to me why `4` minutes and not `e.g. `2` minutes.

Reply:
Yes - I acknowledge that 4 vs 2 is seemingly arbitrary. Let's stick with 4, deferring to the work done to determine that timeout in the Terraform provider.

Ideally, this would be implemented by storing the operation ID in a status condition and then relying on the api machinery to do the backoff, but that doesn't seem worth the extra complexity at the moment.